src/GoogleFit/FitOnFhir.GoogleFit/Services/UsersService.cs

```diff
@@ -111,5 +119,41 @@ public override async Task QueueFitnessImport(User user, CancellationToken cance
                 await _queueService.SendQueueMessage(user.Id, googleUserInfo.UserId, googleUserInfo.PlatformName, cancellationToken);
             }
         }
+
+        public override async Task<bool> RevokeAccess(string patientId, string system, CancellationToken cancellationToken)
+        {
+            var user = await RetrieveUserForPatient(patientId, system, cancellationToken);
+
+            if (user != null)
+            {
+                var platformInfo = user.GetPlatformUserInfo()
+                    .FirstOrDefault(pui => pui.PlatformName == GoogleFitConstants.GoogleFitPlatformName);
+
+                if (platformInfo != default)
+                {
+                    AuthTokensResponse tokensResponse;
+                    try
+                    {
+                        tokensResponse = await _googleFitTokensService.RefreshToken(platformInfo.UserId, cancellationToken);
+                    }
+                    catch (TokenRefreshException ex)
```

Comment:
I think you should just let this throw (not catch) and the calling class will log the exception and return a 500.